engine/hunspell_table.py

```
@@ -10657,18 +10622,29 @@ def _process_key_event(self, key: itb_util_core.KeyEvent) -> bool:
+                    # preedit, the transliteration needs to be
+                    # finalized before committing and then passing
+                    # BackSpace to the application. But finalizing
+                    # with BackSpace does not work for most input
+                    # methods using BackSpace for undo. Use a regular
+                    # space instead to finalize the transliteration.
+                    # See: https://github.com/mike-fabian/ibus-typing-booster/issues/923
+                    final_msymbol = ' '
                 input_phrase = self._transliterators[
                     preedit_ime].transliterate(
-                        self._typed_string + [key.msymbol],
+                        self._typed_string + [final_msymbol],
```

Ruff (0.16.2)

[warning] Consider `[*self._typed_string, final_msymbol]` instead of concatenation

Replace with `[*self._typed_string, final_msymbol]`

(RUF005)